### DISCUSSION TITLE
Disabling programmatically declared extensions

### DISCUSSION THREAD

**Discussion by ghost** [junit-team/junit-framework#3570] [Q&A] [answered]
I have an extension declared using @RegisterExtension annotation and I would like to control whether this extension is enabled.

Is there a way to disable such an extension, for example via an environment variable?

**Comment by sbrannen** [comment, marked as answer]
This is a duplicate of:

- #1242

Please see the comments in that issue for details.

**Comment by ghost** [comment]
Thanks for the quick answer, @sbrannen.

Maybe we should add your example

```java
@RegisterExtension
Extension testWatcher = Boolean.getBoolean("browserstack.enabled") ?
    new BrowserStackTestWatcher() :
    new Extension() {};
```

to the annotation documentation to avoid asking such questions in the future?

**Reply by sbrannen** [reply]
Since that is a bit of a "hack" (relying on knowledge of internal implementation details that may change in the future), we would prefer not to document that as "the solution" to conditional extension registration.

Furthermore, as pointed out in the discussions in #1242, there are multiple ways to have your extension conditionally perform work. So it's better for extension authors to decide on a case-by-case basis how to do so.
